### PR TITLE
refactor(styles): start making things sassy, flexy, responsive

### PR DESCRIPTION
**Pull request by johngruen**
@nchapman: here you go

### CODE REVIEW

**Comment by johngruen** [comment]
@nchapman, i'm gonna have to rebase this once all your stuff gets merged :(

**Comment by 6a68** [comment]
@johngruen mind rebasing? :-)

File: app/styles/_state.scss [hunk 1]
```diff
@@ -9,3 +9,12 @@
 .is-warning {
   color: $warning;
 }
+
+
+@mixin is-border-state-machine($border-width, $base-color, $engaged-color) {
+  transition: all $fast;
+  border: $border-width solid $base-color;
```
Comment (pdehaan) [inline app/styles/_state.scss]: Everything else in your CSS is alphabetized, may as well sort these too.

File: app/styles/modules/_global_header.scss [hunk 2]
```diff
@@ -1,39 +1,57 @@
...
+    line-height: $large-font-size;
+    margin: 0 $default-gap;
   }
 
-  #user {
-    display: none;
+  #user { 
+    background: $baby-grey;
+    border-radius: 50%;
+    font-size: 0px;
```
Comment (pdehaan) [inline app/styles/modules/_global_header.scss]: `font-size: 0;` (no px)

File: app/styles/modules/_global_header.scss [hunk 3]
```diff
@@ -1,39 +1,57 @@
...
-      @include hidpi-background-image(icon-search, 16px 16px);
-      background-position: top 50% right 10px;
-      background-repeat: no-repeat;
-      border: 0;
-      border-radius: 6px;
-      font-size: 14px;
-      padding: 7px 7px 7px 15px;
-      width: 350px;
+    flex-basis: $default-grid * 10;
+      input {
```
Comment (pdehaan) [inline app/styles/modules/_global_header.scss]: Indenting looks "off" here. Me thinks that `input {` should be the same indentation level as `flex-basis`. But whatevah.

File: app/styles/_state.scss (hunk 1 above)
Comment (johngruen) [inline app/styles/_state.scss]: okay

File: app/styles/modules/_global_header.scss [hunk 4]
```diff
@@ -1,39 +1,57 @@
...
+        @include hidpi-background-image(icon-search, 16px 16px);
+        @include color-scheme($middle-grey, $white);
+        @include is-border-state-machine(2px, $middle-grey, $dark-grey);
+        background-position: top 50% right 12px;
+        background-repeat: no-repeat;
+        border-radius: $double-radius;
+        flex: 1;
+        font-size: $default-font-size;
+        padding:  $small-gap $quad-gap $small-gap $small-gap;
```
Comment (pdehaan) [inline app/styles/modules/_global_header.scss]: Extra space after the `:` here... I'm getting really nit-picky, but I'm just basically repeating whatever scss-lint tells me.

File: app/styles/modules/_global_header.scss [hunk 5]
```diff
@@ -1,66 +1,66 @@
...
     }
   }
+}
 
-  #search-box {
-    margin-top: -20px;
-    position: absolute;
-    right: 20px;
-    top: 50%;
+@media screen and (max-width: 520px){
```
Comment (pdehaan) [inline app/styles/modules/_global_header.scss]: Speaking of variables... Want to make the `520px` a variable and tweak it up here and other places that define breakpoints?

File: app/styles/modules/_global_header.scss [hunk 6]
```diff
@@ -1,66 +1,66 @@
...
-    position: absolute;
-    top: 50%;
-
     img {
-      border-radius: 50%;
-      height: 36px;
-      width: 36px;
+      width: $double-grid;
+      height: $double-grid;
+      background: #fafafa;
```
Comment (pdehaan) [inline app/styles/modules/_global_header.scss]: Should this `#fafafa` be a variable?

**Comment by nchapman** [comment]
:+1:

File: app/styles/modules/_user_page.scss [hunk 7]
```diff
@@ -15,48 +15,50 @@
...
-        height: 16px;
-        margin-right: 5px;
-        vertical-align: -3px;
-        width: 16px;
-      }
+    .favicon {
+      height: $half-grid;
+      margin-right: 5px;
+      vertical-align: -3px;
+      width: 1$half-grid;
```
Comment (pdehaan) [inline app/styles/modules/_user_page.scss]: @johngruen Can you confirm this weird looking `width` value here?

Should this be `$half-grid` or was the `1` in there performing some sort of magic?

**Pull request merged**